pkg/manifests/external_dns.go

```diff
@@ -77,21 +78,25 @@ func (rt ResourceType) generateResourceDeploymentArgs() []string {
 
 }
 
-func (rt ResourceType) generateRBACRules() []rbacv1.PolicyRule {
+func (rt ResourceType) generateRBACRules(dnsconfig *ExternalDnsConfig) []rbacv1.PolicyRule {
 	switch rt {
 	case ResourceTypeGateway:
-		return []rbacv1.PolicyRule{
-			{
-				APIGroups: []string{""},
-				Resources: []string{"namespaces"},
-				Verbs:     []string{"get", "watch", "list"},
-			},
+		ret := []rbacv1.PolicyRule{
 			{
 				APIGroups: []string{"gateway.networking.k8s.io"},
 				Resources: []string{"gateways", "httproutes", "grpcroutes"},
 				Verbs:     []string{"get", "watch", "list"},
 			},
 		}
+		if !dnsconfig.isNamespaced {
```

Comment:
surprising that these are the only permissions that change

Reply:
yeah, the broader difference is role vs cluster role